File: app/src/main/java/com/example/fushuang/kt/MyPresenter.java
package com.example.fushuang.kt;

import java.util.ArrayList;

import javax.crypto.spec.IvParameterSpec;

/**
 * Created by fushuang on 2018/3/21.
 */

public class MyPresenter {
    IView view;
    public void attachView(IView iView){
        view=iView;
    }
    public void getData(){


        ArrayList<Cat> cats = new ArrayList<>();
        cats.add(new Cat(1,12));
        cats.add(new Cat(2,13));


        view.showData(cats);
    }
}

File: app/src/main/java/com/example/fushuang/kt/presenter/MyPresenter.kt
package com.example.fushuang.kt.presenter

import android.support.v7.widget.RecyclerView
import com.example.fushuang.kt.view.view.IView
import com.example.fushuang.kt.module.Cat

import kotlin.collections.ArrayList

/**
 * Created by fushuang on 2018/3/21.
 */

class MyPresenter() {


  fun getData(recyclerView: RecyclerView, data: ArrayList<Cat>) {
    data.clear()
    for (i in 1..10) {
      data.add(Cat(1, i))
    }
    recyclerView.adapter.notifyDataSetChanged()

  }
}
